src/System.Runtime.Extensions/src/System/Environment.cs

```diff
@@ -44,7 +45,15 @@ private static Hashtable ToHashtable(this IEnumerable<KeyValuePair<string, strin
             Hashtable hashTable = new Hashtable();
             foreach (KeyValuePair<string, string> pair in pairs)
             {
-                hashTable.Add(pair.Key, pair.Value);
+                try 
+                {
+                    hashTable.Add(pair.Key, pair.Value);                 
+                }
+                catch (Exception ex)
+                {
+                    // Throw and catch intentionally to provide non-fatal notification about corrupted environment block
+                    Debug.WriteLine(SR.Diagnostic_ThrowingException, ex.ToString());
```

Comment:
I do not think we should be writing out anything here. These `Debug.WriteLines` in the core libraries are always super annoying when you hit them.

Just catching the exception is enough.